### ISSUE TITLE
helm lint doesn't accept negative lookahead with regex expressions

### ISSUE THREAD

**Issue by jbilliau-rcd** [helm/helm#9872]
Output of `helm version`: 3.6.1

Output of `kubectl version`: 1.21.0

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  AWS

I have a values.schema.json defined as part of a helm chart that we give developers to deploy their applications. I've recently added a section to do regex testing on the URL's they pass in; we only want to allow domain names, not full URL's with *http://* in front. Thus, I added this to my values.schema.json:

```
"anyOf": [
                {
                  "properties": {
                    "enabled": {
                      "const": true
                  },
                  "hosts": {
                    "additionalProperties": {
                      "pattern": "^(?!http).[a-z1-9]+([0-9]|[a-z]|[-.])+:[0-9]{2,5}$"
                      }
                    }
                  },
                  "required": [
                    "hosts"
                  ]
                }
```

Now, if I go do a `helm lint charts/app-template/ --values values.yaml`, I get:

```
[ERROR] values.yaml: pattern must be a valid regex
[ERROR] templates/: values don't meet the specifications of the schema(s) in the following chart(s):
app-template:
pattern must be a valid regex
```

If I remove the ?, it works (i dont get that error); so it **appears** that helm lint doesnt support regex negative lookahead, that is, excluding a string from the beginning of an expected value.

Basically, I want this to work: www.yahoo.com:443
and this to fail: https://www.yahoo.com:443

I've tried Helm 3.5.4 and 3.6.1, same on both.

**Comment by bacongobbler** [comment]
https://github.com/helm/helm/issues/9651#issuecomment-828611962 explains most of this in detail.

**Comment by jbilliau-rcd** [comment]
Ah ok, missed that. Apologies, looks like it's a json schema issue; argh. Shall I just close this issue then?

**Comment by bacongobbler** [comment]
Yeah. Please file tickets upstream. Thanks!
